app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala

```diff
@@ -187,18 +190,33 @@ class SubscriptionController @Inject()
   val showSubscriptionComplete: Action[AnyContent] = AuthorisedWithSubscribingAgentAsync() {
     implicit authContext =>
       implicit request => {
-        Future successful{
           val agencyData = for {
             agencyName <- request.flash.get("agencyName")
             arn <- request.flash.get("arn")
           } yield (agencyName, arn)
 
-          agencyData.map(data =>
-            Ok(html.subscription_complete(appConfig.agentServicesAccountUrl, data._1, data._2))
-          ) getOrElse sessionMissingRedirect()
-        }
+          sessionStoreService.fetchContinueUrl.map {_ match {
+              case Some(continueUrl) => responseWithOrWithoutContinueUrl(ContinueUrlValue(continueUrl), agencyData)
+              case _ => responseWithOrWithoutContinueUrl(stringData = agencyData)
+            }
+          }
       }
   }
+
+  private def responseWithOrWithoutContinueUrl(continueUrl:String = "", stringData: Option[(String,String)])
+                                              (implicit hc: HeaderCarrier, request: Request[_]) = {
+    val encodedUrl = encodeUtf8(appConfig.agentServicesAccountUrl + continueUrl)
+    stringData.map(data =>
+      Ok(html.subscription_complete(encodedUrl, data._1, data._2))
```

Comment:
You could use real names instead of _1 and _2 here.
```
stringData.map { case (agencyName, arn) =>
      Ok(html.subscription_complete(encodedUrl, agencyName, arn))
    }.getOrElse(sessionMissingRedirect())
```